tooling/ipns/record.go

```diff
@@ -41,10 +51,39 @@ func (i *IpnsRecord) Validity() time.Time {
 }
 
 func (i *IpnsRecord) Valid() error {
-	id, err := peer.Decode(i.key)
+	return ipns.ValidateWithPeerID(i.id, i.pb)
+}
+
+func (i *IpnsRecord) idV1(codec multicodec.Code, base mbase.Encoding) (string, error) {
+	c := peer.ToCid(i.id)
+	c = cid.NewCidV1(uint64(codec), c.Hash())
+	s, err := c.StringOfBase(base)
 	if err != nil {
-		return err
+		return "", err
 	}
+	return s, nil
+}
+
+func (i *IpnsRecord) IntoCID(codec multicodec.Code, base mbase.Encoding) string {
```

Comment:
I'd find the following more natural given all the toStrings, etc. 

```suggestion
func (i *IpnsRecord) ToCID(codec multicodec.Code, base mbase.Encoding) string {
```